cli/src/main/java/org/jboss/as/cli/handlers/ShutdownHandler.java

```diff
@@ -139,25 +162,31 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
         final ModelNode op = this.buildRequestWithoutHeaders(ctx);
 
+        boolean isPerformInstallation = Util.TRUE.equalsIgnoreCase(performInstallation.getValue(ctx.getParsedCommandLine()));
         boolean disconnect = true;
-        final String restartValue = restart.getValue(ctx.getParsedCommandLine());
-        if (Util.TRUE.equals(restartValue) ||
+        final boolean requestRestart = Util.TRUE.equalsIgnoreCase(restart.getValue(ctx.getParsedCommandLine()))
+                || isPerformInstallation;
+        if (requestRestart ||
                 ctx.isDomainMode() &&
                 !isLocalHost(ctx.getModelControllerClient(), host.getValue(ctx.getParsedCommandLine()))) {
             disconnect = false;
         }
 
-        try {
-            final ModelNode response = cliClient.execute(op, true);
-            if(!Util.isSuccess(response)) {
-                throw new CommandLineException(Util.getFailureDescription(response));
-            }
-        } catch(IOException e) {
-            // if it's not connected, it's assumed the connection has already been shutdown
-            if(cliClient.isConnected()) {
-                StreamUtils.safeClose(client);
-                throw new CommandLineException("Failed to execute :shutdown", e);
+        if (Util.TRUE.equalsIgnoreCase(performInstallation.getValue(ctx.getParsedCommandLine()))) {
+            // Check if I am using a local client
+            boolean localClientLaunch = isLocalClientLaunch(ctx);
+            executeOperation(client, cliClient, op, !localClientLaunch);
+            if (localClientLaunch) {
+                try {
+                    TimeUnit.SECONDS.sleep(3);
+                } catch (InterruptedException e) {
+                    // Ignored
+                }
+
+                System.exit(0);
```

Comment:
Isn't the issue more than just the jboss-modules.jar? The CLI process will be using a variety of things in the modules dir.

Reply:
Yes right, I focused my explanation on the `jboss-modules.jar` which is used to launch the modular environment for the CLI process but it also applies to any JAR file used by JBoss Modules as resource loader. The Jars used as resource loaders can be even deleted in Windows but the `jboss-modules.jar` is more strict, Windows locks its deletion I guess because it is used by the JVM as entry point.